dev-portal/src/components/Admin/Accounts/AccountsTable.jsx

```diff
@@ -0,0 +1,331 @@
+        orderDirectionIndex={orderDirectionIndex}
+        setOrderDirectionIndex={setOrderDirectionIndex}
+      />
+      <Table.Body>{tableRows}</Table.Body>
+      <Table.Footer>
+        <Table.Row>
+          <Table.HeaderCell colSpan={columns.length}>
+            <Container fluid textAlign='right'>
+              <Pagination
+                // SemanticUI uses 1-indexing in Pagination. We prefer sanity.
+                activePage={activePage + 1}
+                onPageChange={onPageChange}
+                totalPages={totalPages}
+              />
+            </Container>
+          </Table.HeaderCell>
+        </Table.Row>
+      </Table.Footer>
+    </Table>
+  )
+
+  return (
+    <Container fluid>
+      {toolbar}
+      {table}
+    </Container>
+  )
+}
+
+const TableHeader = React.memo(
+  ({
+    columns,
+    orderColumn,
+    setOrderColumn,
+    orderDirectionIndex,
+    setOrderDirectionIndex,
+  }) => {
+    // Clicking on a column makes it the "orderColumn". If that column was
+    // already the "orderColumn", cycle between order directions (none,
+    // ascending, descending). Otherwise, start at the beginning of the cycle
+    // (ascending).
+    const onToggleOrder = column => () => {
+      if (column === orderColumn) {
+        const nextIndex = nextDirectionIndex(orderDirectionIndex)
+        if (nextIndex === 0) {
+          setOrderColumn(NO_ORDER_COLUMN)
+        }
+        setOrderDirectionIndex(nextIndex)
+      } else {
+        setOrderColumn(column)
+        setOrderDirectionIndex(nextDirectionIndex(0))
+      }
+    }
+
+    const orderDirection = ORDER_DIRECTIONS[orderDirectionIndex]
+    return (
+      <Table.Header>
+        <Table.Row className={styles.headerRow}>
+          {columns.map((column, index) => (
+            <Table.HeaderCell
+              key={index}
+              onClick={column.ordering && onToggleOrder(column)}
+            >
+              {column.title}
+              {column === orderColumn && (
+                <Icon name={orderDirection.iconName} />
+              )}
+              {column.ordering && column !== orderColumn && (
+                <Icon name={ORDER_DIRECTIONS[0].iconName} disabled />
+              )}
+            </Table.HeaderCell>
+          ))}
+        </Table.Row>
+      </Table.Header>
+    )
+  },
+)
+
+const LoadingAccountRow = React.memo(({ columnCount }) => (
+  <Table.Row>
+    {Array.from({ length: columnCount }).map((_value, index) => (
+      <Table.Cell key={index}>
+        <Placeholder data-testid='accountRowPlaceholder' fluid>
+          &nbsp;
+        </Placeholder>
+      </Table.Cell>
+    ))}
+  </Table.Row>
+))
+
+const FillerAccountRow = React.memo(({ columnCount }) => (
+  <Table.Row>
+    {Array.from({ length: columnCount }).map((_value, index) => (
+      <Table.Cell key={index}>&nbsp;</Table.Cell>
+    ))}
+  </Table.Row>
+))
+
+const AccountRow = React.memo(({ account, columns, isSelected, onSelect }) => {
```

Comment:
You probably know this already, but be aware that React.memo only does a shallow compare. e.g. You can't push a new column to the columns array and expect it to rerender.

Reply:
Yep, thanks for the warning though.